pkg/util/error.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Fields is a type alias for a map[string]interface{} to be
+// used to capture key-value pairs to appends to a structured
+// error string
+type Fields map[string]interface{}
+
+// ErrorMessage constructs a structured error string from a
+// message and a set of fields (key-value pairs). The message
+// is formatted as:
+// 		message key1=value1 key2=value2 ...
+func ErrorMessage(message string, context Fields) string {
+	var kvs []string
+
+	for k, v := range context {
+		kvs = append(kvs, fmt.Sprintf("%s=%v", k, v))
+	}
+
+	return fmt.Sprintf("%s %s", message, strings.Join(kvs, " "))
```

Comment:
Hmmm... It would be nice if the fields were exposed to logrus's `WithFields()` method. Maybe we need our own custom `Wrap()` style method that both retains the original error and maintains fields? Or something along those lines. WDYT @davecheney?

Reply:
You're saying have a custom error type that contains a Fields map and when logging an errors the fields get passed to logrus?

Reply:
Yeah, something along those lines. I'm not crazy about creating an error message where the string has the fields hardcoded as key=value pairs. But maybe I'm overthinking/overworrying?

Reply:
I can create a custom logrus formatter to do this (which I think I need to do anyway to get the stack traces to be logged).

Reply:
Before doing that let's see what @davecheney thinks

Reply:
I can definitely write the stack trace to the log using a logrus hook (they allow you to permute the log Entry before it's written, so I can write the error using `%+v` there). This could theoretically work for errors with fields as well (though per earlier discussion, @ncdc does not want to implement this).